plugins/keycloak-backend/src/lib/read.ts

```diff
@@ -66,11 +72,17 @@ export const parseUser = (
   spec: {
     profile: {
       email: user.email,
-      displayName: [user.firstName, user.lastName].filter(Boolean).join(' '),
+      displayName:
+        user.firstName || user.lastName
+          ? `${user.firstName} ${user.lastName}`
+          : user.username,
```

Comment:
How is this related to the merit of this PR? Besides it introduces a regression. If only one of `firstName` or `lastName` is defined, the display name starts/ends with a leading space.

Reply:
@christophe-f and I found another bug. Currently, if a user does not specify a `firstName` and `lastName`, the field will remain empty; therefore, backstage will not inject the entity into the catalog. I will refactor the snippet to not have the regression; otherwise, I can create another pr if needed.